app/src/main/java/lt/vilnius/tvarkau/BaseActivity.java

```diff
@@ -1,19 +1,55 @@
 package lt.vilnius.tvarkau;
 
+import android.app.AlertDialog;
+import android.app.ProgressDialog;
+import android.content.Intent;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.common.api.OptionalPendingResult;
+import com.google.android.gms.common.api.ResultCallback;
 
 import icepick.Icepick;
+import lt.vilnius.tvarkau.utils.SharedPrefsManager;
 
 /**
  * Created by Karolis Vycius on 2016-01-15.
  */
-public abstract class BaseActivity extends AppCompatActivity {
+
+/*
+    This class also is responsible for keeping user google account signed in. It is very important, that all activities, that
+    will be using google sign in, would extend this class.
+
+ */
+public abstract class BaseActivity extends AppCompatActivity  implements GoogleApiClient.OnConnectionFailedListener {
+
+
+    protected GoogleApiClient mGoogleApiClient;
+    public GoogleSignInAccount mGoogleSignInAccount; //Lets us access user information.
+
+    private ProgressDialog mProgressDialog;
+
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         Icepick.restoreInstanceState(this, savedInstanceState);
+
+
+        //Configure google sign in to request user email. Is ID and photo is included in Default Sign In.
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
```

Comment:
Let's keep all Sign-in stuff in a separate activity, not BaseActivity. We'll have to have our own User class and store it locally when user is signed in. 

Or are there a reason why each activity should be able to use g+ sign-in?